### DISCUSSION TITLE
DEV SESSION [TUESDAY] 28 May -- 2030 UTC+8/1430 CEST/1330 BST/ 0830 ET

### DISCUSSION THREAD

**Discussion by jackiekazil** [mesa/mesa#2143] [General]
Where: [via Matrix](https://matrix.to/#/#project-mesa:matrix.org)

Previous meeting: https://github.com/projectmesa/mesa/discussions/2127

**Proposed Agenda**

- General updates
  - GSoC updates 
   - Scheduling, deliverables, etc for advisors 
  - PyCon 2024
- PRs:
  - ???
- #2140
- PropertyLayer and Cell Space visualisation
  - #2137
  - #2138
- Updating the Mesa Examples
  - https://github.com/projectmesa/mesa-examples/issues/111
  - https://github.com/projectmesa/mesa-examples/issues/123
  - https://github.com/projectmesa/mesa-examples/issues/124).
- [Mesa 3.0](https://github.com/projectmesa/mesa/milestone/43)
- Mesa-Geo
- Papers
  - Mesa 3.0 paper -- Summarize updates. (Boyu)
  - Main Mesa citation -- JOSS paper (Jackie)
    - Question from Boyu: authorship

cc @tpike3, @EwoutH, @Corvince , @wang-boyu , @rht, @quaquel, @adamamer20, @harshmahesheka

**Comment by EwoutH** [comment]
Thanks for the agenda! Would a general update about the foundation process be possible?

Integrating the PropertyGrid into the visualization is something I also would like to discuss.

**Reply by jackiekazil** [reply]
What are the ticket numbers associated with the integration?

**Reply by jackiekazil** [reply]
Also feel free to edit the agenda directly.

**Reply by EwoutH** [reply]
#2137 and #2138.

**Reply by EwoutH** [reply]
I added some open issues to the agenda.

**Comment by jackiekazil** [comment]
RE: mesa 3.0 - does this milestone contain everything? https://github.com/projectmesa/mesa/milestone/43

**Reply by EwoutH** [reply]
For 2.3 --> 3.0 it looks like it does. If you start counting from 2.0 some stuff is missing.

**Comment by EwoutH** [comment]
@quaquel do you think you can make it to next week's meeting? If so, it might be nice to continue pushing the data collector forward (#1944).

**Reply by quaquel** [reply]
I might be able to join from 15:00 onwards

**Comment by wang-boyu** [comment]
Unfortunately I won't be able to attend the meeting - will be on a flight back to Singapore at that time. There's probably Internet access available but video meeting isn't allowed.

**Comment by jackiekazil** [comment]
You are correct. This is not allowed. You can skip it with no worries.
Let us know if you have something specific.

**Reply by wang-boyu** [reply]
My main question is regarding the authorship about the JOSS paper - do we keep adding people in the paper or are there better ways of handling it. I have updated the agenda with my questions.

**Comment by rht** [comment]
@jackiekazil should we regularly at-mention the GSoC student for the dev sessions?

**Reply by jackiekazil** [reply]
I added @adamamer20, @harshmahesheka to this. I will add them to the others I created as well. Do you know Dong's github handle?

**Reply by adamamer20** [reply]
i'm truly sorry but i won't be able to join the call today as I have an assignment to deliver by tonight and I'm not finished yet. I will make sure to join on the 4th!

**Comment by jackiekazil** [comment]
Notes from the meeting... 

**GSoC updates**
* I signed up, but Numfocus came in last minute -- they are going to try to take it for us. TBD on where this ends up. 
* Primary advisors Confirmation 
  * @tpike3  -- Dong - Cachable Mesa
  * @jackiekazil -- @harshmahesheka - RL 
  * @rht - Adam (@rht wasn't available to join) 
* GSoC starts this week.
* Meeting -- Let’s meet same time every Tuesday -- unless this doesn’t work for people 

**PyCon 2024**
* Updates on sprint - lower turn out this year, less activity 
* Hoping to do more prework next year -- 
  * Signage? Challenge coins? t-shirt? 
 
**Tutorial - CSSSA** 
* @tpike3 will be taking the work from AGG and reworking. He is focused on making Jupyter lite work. 
* Tutorial is on June 5

**Mesa 3.0** 
* https://github.com/projectmesa/mesa/milestone/43
* Targeting end of the summer -- before next year.
* @EwoutH, it was mentioned you have a timeline for this? Is this documented or just verbally communicated? Or is it not something that really exists, and people just think it exists? 

**Paper discussion**
Let's do two papers? Maybe three? Papers all around. 
* Mesa-geo - Do we need an update for this? 
* Mesa 3.0 updates - Simulation
* Mesa --- general - mini - JOSS

**Future of Data Collection Update #1944**
* https://github.com/projectmesa/mesa/discussions/1944 
* @quaquel had some personal delays (we all do) --- he will get back to it in June. 
* Suggested route: Make an experimental implementation -- so it can be pulled out if needed.

I am sure I missed something, but I think this covers most of it.
